reactive/kubernetes_master.py

```diff
@@ -1289,8 +1289,10 @@ def create_tokens_and_sign_auth_requests():
             continue
         kubelet_token = get_token(username)
         if not kubelet_token:
-            # Usernames have to be in the form of system:node:<nodeName>
-            userid = "kubelet-{}".format(request[0].split('/')[1])
+            # Username will be in the form of system:node:<nodeName>.
+            # User ID will be a worker <unitName>, and while not used today, we store
+            # this in case it becomes useful to map a secret to a unit in the future.
+            userid = request[0]
```

Comment:
I'm starting to think we could save a lot of confusion if we just use the username as the uid in all cases, i.e. instead of

```
username: system:node:ip-172-31-37-122
uid: kubernetes-worker/0
```

we use the same for both:

```
username: system:node:ip-172-31-37-122
uid: system:node:ip-172-31-37-122
```

Reply:
I mention this because `uid: kubernetes-worker/0` raises questions:

1. Is the `/` safe?
2. What happens if we relate to a kubernetes-worker application in a different model? Will we end up with multiple users that have the same `kubernetes-worker/0` UID?

Reply:
It would potentially be less confusing to make uid==username, but we use the uid in kubeconfig tokens; it's nice to me to see the unit name in there so i know at-a-glance that a kubeconfig goes with a $unit vs a $hostname.

> 1. Is the `/` safe?

Looks safe. We expose the `$unit` as part of a token when validating secrets, and that works fine:

```bash
$ curl -X POST -H "Content-Type: application/json" -d '{"kind": "TokenReview", "apiVersion": "authentication.k8s.io/v1beta1", "spec": {"token": "worker-a/3::r0DzvhNmcWqDLtP72l8Ux2ZJNm5DvKJp"}}' https://10.246.153.172:5000/v1beta1
{"apiVersion":"authentication.k8s.io/v1beta1","kind":"TokenReview","spec":{"token":"worker-a/3::r0DzvhNmcWqDLtP72l8Ux2ZJNm5DvKJp"},"status":{"authenticated":true,"user":{"groups":["system:nodes"],"uid":"worker-a/3","username":"system:node:juju-2a8f8a-15"}}}
```

> 2. What happens if we relate to a kubernetes-worker application in a different model? Will we end up with multiple users that have the same `kubernetes-worker/0` UID?

Yes, we'll have multiple users with the same uid, but they will have unique secret names, and the `auth-webhook` doesn't care about uid, so it'll keep looping through all secrets until it finds a match.